### PR TITLE
Update documentation

### LINKED ISSUE #59511: Background color obstructs ListView hover effect
## Steps to reproduce

Run the following app:

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(FailingListTileHover());
}

class FailingListTileHover extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: DecoratedBox(
          decoration: BoxDecoration(
            color: Colors.white,
          ),
          child: ListView(
            children: [
              Material(
                child: ListTile(
                  title: Text('Enabled; has Material ancestor'),
                  hoverColor: Colors.red,
                  enabled: true,
                  onTap: () {},
                ),
              ),
              Material(
                child: ListTile(
                  title: Text('Disabled; has Material ancestor'),
                  hoverColor: Colors.red,
                  enabled: false,
                  onTap: () {},
                ),
              ),
              ListTile(
                title: Text('Enabled; does not have Material ancestor'),
                hoverColor: Colors.red,
                enabled: true,
                onTap: () {},
              ),
              ListTile(
                title: Text('Disabled; does not have Material ancestor'),
                hoverColor: Colors.red,
                enabled: false,
                onTap: () {},
              ),
            ],
          ),
        ),
      ),
    );
  }
}
```

## Expected behavior

You expect the hover effect (red color) to work on all enabled list tiles (in the example above, the first and the third).

## Actual behavior

Only the list tile that's wrapped in a `Material` widget displays the hover effect.  The list tile that's _not_ wrapped in a `Material` widget does not show the red hover, presumably because the box decoration background is obscuring the hover effect.

This can be seen at https://dartpad.dev/0d2cdf67ab201445667b4d9230844234

### PR DESCRIPTION
**Pull request by tvolkert**
## Description

This is an assortment of documentation updates that largely answers questions I had while building a Flutter app in the respective places I initially looked for the information.

## Related issues

Fixes https://github.com/flutter/flutter/issues/59511

## Tests

n/a

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

### CODE REVIEW

**Comment by tvolkert** [comment]
Fixes https://github.com/flutter/flutter/issues/59511

File: packages/flutter/lib/src/material/outline_button.dart [hunk 1]
```diff
@@ -48,6 +48,22 @@ const Duration _kElevationDuration = Duration(milliseconds: 75);
 /// Outline buttons have a minimum size of 88.0 by 36.0 which can be overridden
 /// with [ButtonTheme].
 ///
+/// {@tool dartpad --template=stateless_widget_scaffold_center}
+///
+/// Here is an example of a basic `OutlineButton`.
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/outline_button.dart:53]: nit:

```suggestion
/// Here is an example of a basic [OutlineButton].
```

File: packages/flutter/lib/src/rendering/box.dart [hunk 2]
```diff
@@ -1445,6 +1449,11 @@ abstract class RenderBox extends RenderObject {
   ///
   /// When the incoming argument is not finite, then they should return the
   /// actual intrinsic dimensions based on the contents, as any other box would.
+  ///
+  /// See also:
+  ///
+  ///  * [computeMaxIntrinsicWidth], which computes smallest width beyond which
```
Comment (goderbauer) [inline packages/flutter/lib/src/rendering/box.dart:1455]: nit: should it be: ... which computes the smallest width ... ?

File: packages/flutter/lib/src/rendering/box.dart [hunk 3]
```diff
@@ -1600,6 +1609,8 @@ abstract class RenderBox extends RenderObject {
   /// See also:
   ///
   ///  * [computeMinIntrinsicWidth], which has usage examples.
+  ///  * [computeMaxIntrinsicHeight], which computes smallest height beyond
```
Comment (goderbauer) [inline packages/flutter/lib/src/rendering/box.dart:1612]: same.

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**